src/mpc-hc/MainFrm.cpp

```diff
@@ -1871,6 +1879,11 @@ void CMainFrame::OnTimer(UINT_PTR nIDEvent)
                     m_wndSubresyncBar.SetTime(rtNow);
                     m_wndSubresyncBar.SetFPS(m_pCAP->GetFPS());
                 }
+
+                const CAppSettings& s = AfxGetAppSettings();
+                if (s.bAutoUploadSubtitles && (rtNow / rtDur == 90) && !m_pSubStreams.IsEmpty() && s.fEnableSubtitles && m_pCAP && m_pCAP->GetSubtitleDelay() == 0) {

```

Comment:
> (rtNow / rtDur == 90)

This doesn't look right. First of all I think it will be true multiple times due to rounding. So you will upload subtitles multiple times. Second it will upload subtitles always at this time period. 

The idea behind "90%" is to ensure that subtitles are correctly synced. If user watched 90% we can assume that subs are valid. Now the key word is "user watched" that doesn't mean user skipped move to the end. 

That say I think you need to add another value which would store amount of time when this subtitles were enabled, possibly with correlation with rtNow to ensure that user watched movie and not lopped first 10 minutes for 2 hours. And when this new value is higher or equal 0.9*rtDur subs should be uploaded. But only once so after first it should abort even trying.

We need to make it sane, because if mpc-hc will start doing mess we will get banned in no time.